osu.Game/OsuGame.cs

```diff
@@ -310,6 +314,12 @@ private void load(FrameworkConfigManager frameworkConfig)
         /// <param name="beatmapId">The beatmap to show.</param>
         public void ShowBeatmap(int beatmapId) => waitForReady(() => beatmapSetOverlay, _ => beatmapSetOverlay.FetchAndShowBeatmap(beatmapId));
 
+        /// <summary>
+        /// Show Direct with a query.
+        /// </summary>
+        /// <param name="query">The query to search for.null</param>
+        public void ShowDirectWithSearch(string query) => waitForReady(() => direct, _ => direct.ShowWithSearch(query));
```

Comment:
to be consistent with other functions naming-side, this should be renamed to `SearchBeatmap` or similar.